.github/workflows/main.yml

```diff
@@ -7,89 +7,55 @@ on:
   workflow_dispatch:
   merge_group:
     types: [checks_requested]
-    
+
 jobs:
-  linux-ci:
-    name: Linux
-    runs-on: ubuntu-latest
+  test:
+    name: ${{ format('{0} {1}', matrix.platform.target, matrix.features)}}
+    runs-on: ${{ matrix.platform.os }}
+    env:
+      RUST_BACKTRACE: 1
     strategy:
       matrix:
-        features: ["", "force-inprocess", "memfd", "async"]
+        platform:
+          - { target: aarch64-apple-darwin, os: macos-14 }
+          - { target: x86_64-apple-darwin, os: macos-13 }
+          - { target: x86_64-unknown-linux-gnu, os: ubuntu-latest }
+          - { target: x86_64-pc-windows-msvc, os: windows-latest }
+          - { target: i686-pc-windows-msvc, os: windows-latest }
+        features: ["", "force-inprocess", "async"]
+        include:
+          - features: "windows-shared-memory-equality"
+            platform: { target: x86_64-pc-windows-msvc, os: windows-latest }
+          - features: "windows-shared-memory-equality"
+            platform: { target: i686-pc-windows-msvc, os: windows-latest }
+          - features: "memfd"
+            platform: { target: x86_64-unknown-linux-gnu, os: ubuntu-latest }
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
 
-      - name: Install nightly toolchain
-        uses: dtolnay/rust-toolchain@nightly
+      - name: Install stable toolchain
+        uses: dtolnay/rust-toolchain@stable
         with:
           components: rustfmt, clippy
 
       - name: rustfmt
         run: cargo fmt --check
 
       - name: clippy
-        run: cargo clippy --features "${{ matrix.features }}"
-
-      - name: Cargo build
-        run: cargo build --features "${{ matrix.features }}"
+        run: cargo clippy --features "${{ matrix.features }}" --target ${{ matrix.platform.target }}
 
       - name: Cargo test
-        run: cargo test --features "${{ matrix.features }}"
-        env:
-          RUST_BACKTRACE: 1
-
-  mac-ci:
-    name: macOS
-    runs-on: macos-latest
-    strategy:
-      matrix:
-        features: ["", "force-inprocess", "async"]
-
-    steps:
-      - uses: actions/checkout@v3
-
-      - name: Install nightly toolchain
-        uses: dtolnay/rust-toolchain@nightly
-
-      - name: Cargo build
-        run: cargo build --features "${{ matrix.features }}"
+        run: cargo test --features "${{ matrix.features }}" --target ${{ matrix.platform.target }}
 
       - name: Cargo test
```

Comment:
I suppose we can rename this to Cargo bench. And maybe we can add some condition like only run on main branch instead of every commit during PR reviewing.

Reply:
> And maybe we can add some condition like only run on main branch instead of every commit during PR reviewing.

This only runs each bench function once to test that it works, so it's blazingly fast.